qtconsole/usage.py

```diff
@@ -0,0 +1,221 @@
+up or down arrow keys, the console will clear the cell and replace it with the
+contents of the one above or below that you navigated to.  If this was an
+accident and you want to retrieve the cell you were editing, use the Undo
+keybinding, ``Control-z``.
+
+
+Key bindings
+============
+
+The Jupyter QtConsole supports most of the basic Emacs line-oriented keybindings,
+in addition to some of its own.
+
+The keybinding prefixes mean:
+
+- ``C``: Control
+- ``S``: Shift
+- ``M``: Meta (typically the Alt key)
+
+The keybindings themselves are:
+
+- ``Enter``: insert new line (may cause execution, see above).
+- ``C-Enter``: *force* new line, *never* causes execution.
+- ``S-Enter``: *force* execution regardless of where cursor is, no newline added.
+- ``Up``: step backwards through the history.
+- ``Down``: step forwards through the history.
+- ``S-Up``: search backwards through the history (like ``C-r`` in bash).
+- ``S-Down``: search forwards through the history.
+- ``C-c``: copy highlighted text to clipboard (prompts are automatically stripped).
+- ``C-S-c``: copy highlighted text to clipboard (prompts are not stripped).
+- ``C-v``: paste text from clipboard.
+- ``C-z``: undo (retrieves lost text if you move out of a cell with the arrows).
+- ``C-S-z``: redo.
+- ``C-o``: move to 'other' area, between pager and terminal.
+- ``C-l``: clear terminal.
+- ``C-a``: go to beginning of line.
+- ``C-e``: go to end of line.
+- ``C-u``: kill from cursor to the begining of the line.
+- ``C-k``: kill from cursor to the end of the line.
+- ``C-y``: yank (paste)
+- ``C-p``: previous line (like up arrow)
+- ``C-n``: next line (like down arrow)
+- ``C-f``: forward (like right arrow)
+- ``C-b``: back (like left arrow)
+- ``C-d``: delete next character, or exits if input is empty
+- ``M-<``: move to the beginning of the input region.
+- ``M->``: move to the end of the input region.
+- ``M-d``: delete next word.
+- ``M-Backspace``: delete previous word.
+- ``C-.``: force a kernel restart (a confirmation dialog appears).
+- ``C-+``: increase font size.
+- ``C--``: decrease font size.
+- ``C-M-Space``: toggle full screen. (Command-Control-Space on Mac OS X)
+
+The pager
+=========
+
+The Jupyter QtConsole will show long blocks of text from many sources using a
+builtin pager. You can control where this pager appears with the ``--paging``
+command-line flag:
+
+- ``inside`` [default]: the pager is overlaid on top of the main terminal. You
+  must quit the pager to get back to the terminal (similar to how a pager such
+  as ``less`` or ``more`` works).
+
+- ``vsplit``: the console is made double-tall, and the pager appears on the
+  bottom area when needed.  You can view its contents while using the terminal.
+
+- ``hsplit``: the console is made double-wide, and the pager appears on the
+  right area when needed.  You can view its contents while using the terminal.
+
+- ``none``: the console never pages output.
```

Comment:
This isn't very clear: it presumably means that long text that would be paged is displayed like output within the flow of the console, but it could suggest that that output is just discarded.